### ISSUE TITLE
share config never works and always reports "ESLint couldn't find the config" in v6

### ISSUE THREAD

**Issue by lwr** [eslint/eslint#12654]
**Tell us about your environment**

We are upgrading ESLint from 4 to 6

* **ESLint Version:** 6.7.2
* **Node Version:** 13.2.0
* **npm Version:** 6.13.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

no config, just make a simple example with standard

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
mkdir -p 1/1 && echo '{}' > 1/package.json
npm i --prefix=1 -D eslint eslint-config-standard eslint-plugin-standard eslint-plugin-import eslint-plugin-node eslint-plugin-promise

echo '{"extends":"standard"}' > 1/.eslintrc
touch 1/1/foo.js

1/node_modules/.bin/eslint --resolve-plugins-relative-to 1 1/1 && echo ok
## output is "ok"

mkdir 2
echo '{"extends":"standard"}' > 2/.eslintrc
touch 2/foo.js
1/node_modules/.bin/eslint --resolve-plugins-relative-to 1 2 && echo ok
## failed with "ESLint couldn't find the config "standard" to extend from"
```

**What did you expect to happen?**
It works in eslint 4 and now broken

**What actually happened? Please include the actual, raw output from ESLint.**
It is broken now in eslint 6, failed with
```
Oops! Something went wrong! :(

ESLint: 6.7.2.

ESLint couldn't find the config "standard" to extend from. Please check that the name of the config is correct.

The config "standard" was referenced from the config file in "/path/to/2/.eslintrc".

If you still have problems, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

option `--resolve-plugins-relative-to` did not do any help
and I can not find any workaround or replaced to make it work again

**Comment by mysticatea** [comment]
Thank you for your report.

This is the correct behavior.
The `2/.eslintrc` uses `eslint-config-standard`, but `require("eslint-config-standard")` in `2/.eslintrc` throws "module not found" error. Therefore, ESLint said "ESLint couldn't find the config "standard" to extend from."

**Comment by lwr** [comment]
@mysticatea thank you for reply

so how to ref a config outside (but not using relative path) correctly in eslint6?

what is the replacement for this using case?

**Comment by lwr** [comment]
I have do a second try

```bash
echo '{"extends":"plugin:es/no-2019","plugins":["es"]}' > 1/.eslintrc
1/node_modules/.bin/eslint --resolve-plugins-relative-to 1 1/1 && echo ok
## outputs "ok"

echo '{"extends":"plugin:es/no-2019","plugins":["es"]}' > 2/.eslintrc
1/node_modules/.bin/eslint --resolve-plugins-relative-to 1 2 && echo ok
## outputs "ok"
```

so it seems that `--resolve-plugins-relative-to` can workaround it by only using plugin instead of config module

but three problems
1. It requires to change or wrap the origin shared config module with an empty plugin
2. eslintrc file  is now more complicated
   - origin
      `extends: x` / `extends: x/foo` / `extends: x/bar`
   - now 
        ```json
        "extends" : "plugin:wrapX/recommended",
        "plugins" : ["wrapX"],
        ```
        ```json
        "extends" : "plugin:wrapX/foo",
        "plugins" : ["wrapX"],
        ```
        ```json
        "extends" : "plugin:wrapX/bar",
        "plugins" : ["wrapX"],
        ```
3. yaml can not exported by plugin, so all the origin config must rewrite

**Comment by mysticatea** [comment]
Generally, we cannot separate config files from dependencies. Those should be tied.

I think that you can use `--config` option to use config files of another place.

    rm 2/.eslintrc
    1/node_modules/.bin/eslint --resolve-plugins-relative-to 1 --config 1/.eslintrc 2 && echo ok

**Comment by lwr** [comment]
I have think about using --config, but it breaks the ci process which run eslint to check root path

we have to split the lint task to run multiple times for different directories and it is difficult

because different directories can have different eslintrc

**Comment by lwr** [comment]
maybe we need somethings liked `--resolve-config-relative-to`

**Comment by lwr** [comment]
Good news, we had just do a success migration in https://github.com/Mailtech/eslint-config-coremail

steps bellow
1. change the module from config to plugin: https://github.com/Mailtech/eslint-config-coremail/commit/7805f2cc29c78cd8591aef2fbe49f21441231a70
2. migrate all existing eslintrc's "extends" property, because no plugin rules, it it no need to add "plugins" property
   -  `extends : ["coremail"]` ->
      `extends : ["plugin:coremail/standard"]`
   -  `extends : coremail/config/compliant.yaml` ->
      `extends : "plugin:coremail/compliant"`
3. change the ci scripts, add `--resolve-plugin-relative-to` option
